### PR TITLE
Thpres refactor

### PR DESCRIPTION
**Pull request by joakim-hove**
Use new 3D property for THPRES initialization

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp [hunk 1]
```diff
@@ -89,13 +89,20 @@ namespace Opm {
...
+            if( !eclipseProperties.hasDeckIntGridProperty("EQLNUM"))
                 throw std::runtime_error("Error when internalizing THPRES: EQLNUM keyword not found in deck");
 
-            //Find max of eqlnum
             const auto& eqlnumKeyword = eclipseProperties.getIntGridProperty( "EQLNUM" );
             const auto& eqlnum = eqlnumKeyword.getData();
-            maxEqlnum = *std::max_element(eqlnum.begin(), eqlnum.end());
+#endif
+            //Find max of eqlnum
+            int maxEqlnum = *std::max_element(eqlnum.begin(), eqlnum.end());
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp]: This dereferences `eqlnum.end()` if `eqlnum` is empty.  Are we guaranteed that that does not happen?

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp]: The `eqlnum` data is default initialized to guarantee that safely.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**